sources/api/apiserver/src/server/mod.rs

```diff
@@ -352,6 +353,27 @@ async fn get_configuration_files(
     Ok(ConfigurationFilesResponse(resp))
 }
 
+/// Reboots the machine
+async fn reboot() -> Result<HttpResponse> {
+    debug!("Rebooting in 10 seconds");
+    // Since there's no logind support, '/sbin/shutdown' cannot schedule reboot and will immediately reboot if called directly.
+    // There is a chance that we won't be able to return a response code in time before being terminated.
+    // Currently relying on systemd to add a relay before the reboot.
+    ensure!(
+        Command::new("/bin/systemd-run")
+            // Add a 10 second delay to shutdown
```

Comment:
What's the reasoning for a 10 second delay?  Service ordering constraints should take care of the logical flow of shutdown, so if the only thing we care about here is having enough time to return an HTTP response code, we only need a tiny fraction of this.  Did you test and find that we do require a delay at all?

Reply:
The 10 seconds is arbitrary and I'm OK with changing it to any value that makes sense. I haven't tested directly calling `shutdown -r`, testing that now.